sim/mage/combustion.go

```diff
@@ -113,4 +114,50 @@ func (mage *Mage) registerCombustionSpell() {
 			spell.Dot(target).Apply(sim)
 		},
 	})
+
+	combustionTickCount := 0
+	combustionTickDamage := 0.0
+	updateCombustionTickCountEstimate := func() {
+		combustionTickCount = int(mage.Combustion.RelatedDotSpell.Dot(mage.CurrentTarget).ExpectedTickCount())
+	}
+	updateCombustionTickDamageEstimate := func(sim *core.Simulation) {
+		combustionTickDamage = mage.Combustion.RelatedDotSpell.ExpectedTickDamage(sim, mage.CurrentTarget)
+	}
+
+	updateCombustionTotalDamageEstimate := func(sim *core.Simulation) int32 {
+		combustionDotDamage := int32(float64(combustionTickCount) * combustionTickDamage)
+
+		if combustionDotDamage != mage.combustionDotEstimate {
+			mage.combustionDotEstimate = combustionDotDamage
+			if sim.Log != nil {
+				mage.Log(sim, "Combustion Dot Estimate: %d - %d Ticks ", combustionDotDamage, combustionTickCount)
+			}
+		}
+
+		return combustionDotDamage
+	}
+
+	mage.AddOnCastSpeedChanged(func(sim *core.Simulation, old float64, new float64) {
+		if sim != nil {
```

Comment:
Basically anytime you have to write a check like this, it means you should rethink how the code is being structured. Callbacks like this should never be called with `nil` inputs or you're just asking for developer headaches and weird bugs down the road.